packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -165,6 +165,13 @@ export function isAssignable(
 	ctx: core.CheckerContext,
 	isEquivalent?: NodeEquivalenceChecker,
 ): boolean {
+	if (
+		assignValue.kind === 'literal' && typeDef.kind === 'literal' &&
+		assignValue.value.kind === typeDef.value.kind &&
+		!assignValue.attributes && !typeDef.attributes
+	) {
+		return assignValue.value.value === typeDef.value.value
```

Comment:
Maybe if this is false, only return if both kinds do not have any attributes set? Or alternatively, call the appropiate mapping attribute function here as well?